### PR TITLE
Add GitHub Actions workflow

### PR DESCRIPTION
**Pull request by bazay**
Remove Travis and replace with GHA

Tested in my local repo:
https://github.com/bazay/rspec-its/actions/runs/421361717

### CODE REVIEW

**Review by pirj — APPROVED** [review]
Cool! 👏 
@JonRowe ?

File: .github/workflows/ruby_ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Ruby CI
+
+on:
+  push:
+    branches:
+      - '**'
```
Comment (JonRowe) [inline .github/workflows/ruby_ci.yml:6]: You don't want to run on all branches that causes double builds.

File: .github/workflows/ruby_ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        ruby-version: [2.7.2, 2.6.6, 2.5.8, 2.4.10, 2.3.8, 2.2.10, 2.1.9]
```
Comment (JonRowe) [inline .github/workflows/ruby_ci.yml:19]: Matrix also needs to account for the builds required to test different rspec versions.

**Review by JonRowe — CHANGES_REQUESTED** [review]
This workflow doesn't replicate our travis setup, so it needs tweaks. It also needs the travis file restoring for Ruby <  2.1

File: .github/workflows/ruby_ci.yml (hunk 2 above)
Comment (bazay) [reply .github/workflows/ruby_ci.yml:19]: GHA doesn't support lesser versions, so I'll re-instate a partial travis file as you suggested

Comment (pirj) [reply .github/workflows/ruby_ci.yml:19]: Can we omit patch version, i.e. use 2.7, 2.6, 2.5, 2.4, 2.3? https://github.com/rspec/rspec-core/blob/main/.github/workflows/ci.yml#L20

Comment (JonRowe) [reply .github/workflows/ruby_ci.yml:19]: To be clear, I mean the matrix of RSpec versions against 2.7, not the matrix of Ruby versions, thats fine.

File: .github/workflows/ruby_ci.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
```
Comment (JonRowe) [inline .github/workflows/ruby_ci.yml:20]:
```suggestion
      matrix:
        ruby:
          - 3.0.0-preview2
          - 2.7
          - 2.6
          - 2.5
          - 2.4
          - 2.3
          - 2.2
          - 2.1.9
        env:
          -
            BRANCH: "main"
        include:
          - ruby: jruby-9.2.13.0
            env:
              JRUBY_OPTS: "--dev"
          - ruby: jruby-9.1.17.0
            bundler: 1
            os: ubuntu-18.04
            env:
              JRUBY_OPTS: "--dev"
          - ruby: 2.7
            env:
              BRANCH: "3-9-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-8-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-7-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-6-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-5-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-4-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-3-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-2-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-1-maintenance"
          - ruby: 2.7
            env:
              BRANCH: "3-0-maintenance"
```

File: .github/workflows/ruby_ci.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    tags-ignore:
+      - '*.*'
+  pull_request:
+    branches:
+      - '**'
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
```
Comment (JonRowe) [inline .github/workflows/ruby_ci.yml:16]:
```suggestion
runs-on: ${{ matrix.os || 'ubuntu-latest' }
```

File: .github/workflows/ruby_ci.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+on:
+  push:
+    branches:
+      - '**'
+    tags-ignore:
+      - '*.*'
+  pull_request:
+    branches:
+      - '**'
```
Comment (JonRowe) [inline .github/workflows/ruby_ci.yml:11]:
```suggestion
on:
  push:
    branches:
      - 'main'
      - '*-maintenance'
      - '*-dev'
  pull_request:
    branches:
      - '*'
```

File: .travis.yml [hunk 6]
```diff
@@ -1,66 +1,60 @@
...
-matrix:
+  - ruby-head	
+  - ree	
+  - jruby-9.2.7.0	
+  - jruby-head	
+  - jruby-1.7.27	
+  - rbx-3	
+env:	
+  - JRUBY_OPTS='--dev'	
+matrix:
```
Comment (JonRowe) [inline .travis.yml:23]:
```suggestion
  - ree	
  - rbx-3	
```

File: .github/workflows/ruby_ci.yml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+          - 2.6
+          - 2.5
+          - 2.4
+          - 2.3
+          - 2.2
+          - 2.1.9
+          - ruby-head
+        include:
+          - ruby: 'jruby-9.2.13.0'
+            env:
```
Comment (bazay) [inline .github/workflows/ruby_ci.yml:30]: It appears GHA doesn't like `env` being specified here. Apparently it's only supported in the `step` which isn't very helpful for us...

Comment (bazay) [reply .github/workflows/ruby_ci.yml:31]: @JonRowe are you sure the GHA flow in `rspec-core` is working?

**Comment by pirj** [comment]
@JonRowe This might be closed?

**Comment by JonRowe** [comment]
@bazay I generated (well, copied from rspec-dev) a fresh github action workflow and added the matrix entries and its up and working, so this can be closed.

**Pull request closed without merging**